services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,75 @@
 package org.openkilda.functionaltests.spec.logging
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def tagName = "kilda-floodlight"
+        def result = elasticService.getLogs(new ElasticQueryBuilder().setTags(tagName).setTimeRange(300).build())
+        assert !result.timedOut
+        assert result?.hits
+        assert result.hits?.total > 0
+
+        then: "There should be discovery messages"
+        def hits = result.hits.hits
+        hits.any {hit -> hit.source.message.toLowerCase().contains("push discovery package via")}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        setup: "Request a non-existent flow"
+        def flowId = "nonexistentflowid" + System.currentTimeMillis()
+        try {
+            northbound.getFlow(flowId)
+        } catch (e) {
```

Comment:
you should catch HttpClientErrorException, not a generic `e` in order to not catch-ignore not-related problems. Also you may want to only ignore on 404 status code in the exception and rethrow for 5xx etc., but this may be an overkill

Reply:
Fixed